validation/spec.json

```diff
@@ -1,68 +1,63 @@
 {
-	"$schema": "http://json-schema.org/draft-06/schema#",
-
-  "definitions": {
-    "channel": {
-      "type": "object",
-      "patternProperties": {
-        "^[a-zA-Z]+$": {
-          "$ref": "#/definitions/sample"
-        }
-      },
-      "additionalProperties": false
-    },
-    "sample": {
-      "type": "object",
-      "properties": {
-        "data": { "$ref": "#/definitions/data"},
-        "mods": {
-          "type": "array",
-          "items": {"$ref": "#/definitions/modification"}
+    "$schema": "http://json-schema.org/draft-06/schema#",
+    "type": "object",
+    "patternProperties": {
+        "^[a-zA-Z0-9]+$": {
+            "$ref": "#/definitions/channel"
         }
-      },
-      "required": ["data", "mods"],
-      "additionalProperties": false
-    },
-    "modification": {
-      "type": "object",
-      "properties": {
-        "name": {
-          "type": "string"
-        },
-        "type": {
-          "type": "string",
-          "enum": ["shapesys","normfactor","histosys","normsys"]
-        },
-        "data": { "$ref": "#/definitions/data"}
-      },
-      "additionalProperties": false
     },
-    "data": {
-      "oneOf": [
-        {
-          "type": "array",
-          "items": {"type": "number"}
+    "additionalProperties": false,
+
+    "definitions": {
+        "channel": {
+            "type": "object",
+            "properties": {
+                "samples": { "type": "array", "items": {"$ref": "#/definitions/sample"} }
+            },
+            "additionalProperties": false
         },
-        {
-          "type": "null"
+        "sample": {
+            "type": "object",
+            "properties": {
+                "name": { "type": "string" },
+                "data": { "$ref": "#/definitions/data" },
+                "modifiers": { "type": "array", "items": {"$ref": "#/definitions/modifier"} }
+            },
+            "required": ["name", "data", "modifiers"],
+            "additionalProperties": false
         },
-        {
-          "type": "object",
-          "required": ["lo_hist", "hi_hist"]
+        "modifier": {
+            "type": "object",
+            "properties": {
+                "name": {
+                    "type": "string"
+                },
+                "type": {
+                    "type": "string",
+                    "enum": ["shapesys","normfactor","histosys","normsys"]
+                },
+                "data": { "$ref": "#/definitions/data"}
+            },
+            "additionalProperties": false
         },
-        {
-          "type": "object",
-          "required": ["lo", "hi"]
+        "data": {
+            "oneOf": [
+                {
```

Comment:
should we maybe group those definitions into it's separate location in the definitions and have something like

```yaml
oneOf:
- {$ref: #/definitions/modifier_definitions/histosysdata}
- {$ref: #/definitions/modifier_definitions/normsysdata}
- {$ref: #/definitions/modifier_definitions/shapesysdata}
```

Reply:
Yeah, let's file an issue for this, as this is out of the scope of fully-specified channels/samples.